File: up.js
const { execSync } = require('child_process');

function runCommand(command) {
  try {
    const output = execSync(command).toString();
    return output;
  } catch (error) {
    console.error(error.stderr.toString());
  }
}

async function gitPush(commitMessage) {
  try {
    runCommand('git add *');
    runCommand(`git commit -m "${commitMessage}"`);
    const output = runCommand('git push');
    console.log(
      output.includes('Everything up-to-date')
        ? 'Tidak ada perubahan'
        : 'Berhasil'
    );
  } catch (error) {
    console.error(error);
  }
}

async function main() {
  const commitMessage = await new Promise((resolve) => {
    process.stdout.write('Masukkan pesan commit: ');
    process.stdin.resume();
    process.stdin.on('data', (data) => {
      resolve(data.toString().trim());
    });
  });
  await gitPush(commitMessage);
}

main();

File: up.py
import subprocess

def run_cmd(cmd):
    return subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True).communicate()[0].decode()

def git_push(commit_message):
    run_cmd("git add *")
    run_cmd(f'git commit -m "{commit_message}"')
    output = run_cmd("git push")
    if "Everything up-to-date" in output:
        print("Tidak ada perubahan")
    elif "Everything up-to-date" not in output:
        print("Berhasil")

if __name__ == "__main__":
    commit_message = input("Masukkan pesan commit: ")
    git_push(commit_message)
